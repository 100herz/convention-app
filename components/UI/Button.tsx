import React from 'react'
import { GestureResponderEvent, StyleSheet, TextStyle, View, ViewStyle } from 'react-native'

import Touchable from '@components/UI/Touchable'
import Text from '@components/UI/Text'
import { colors, fonts } from '@styles/theme'

interface Props {
  onPress: (event: GestureResponderEvent) => void
  title: string
  style?: ViewStyle
  textStyle?: TextStyle
}

const Button: React.FC<Props> = ({ onPress, title, style, textStyle }) => {
  return (
    <Touchable onPress={onPress}>
      <View style={{ ...styles.container, ...style }}>
        <Text style={{ ...styles.text, ...textStyle }}>{title}</Text>
      </View>
    </Touchable>
  )
}

interface Styles {
  container: ViewStyle
  text: TextStyle
}

const styles = StyleSheet.create<Styles>({
  container: {
    backgroundColor: colors.accentColor,
    paddingHorizontal: 15,
    paddingVertical: 2,
    borderRadius: 4,
  },
  text: {
    color: 'white',
    textTransform: 'uppercase',
    fontFamily: fonts.sansBold,
  },
})

export default Button
