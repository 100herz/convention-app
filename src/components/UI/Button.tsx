import React from 'react'
import { GestureResponderEvent, StyleSheet, TextStyle, View, ViewStyle } from 'react-native'

import Touchable from '@components/UI/Touchable'
import Text from '@components/UI/Text'
import { colors, fonts } from '@styles/theme'

interface Props {
  onPress: (event: GestureResponderEvent) => void
  title: string
  style?: ViewStyle
  textStyle?: TextStyle
  testID?: string
}

const Button: React.FC<Props> = ({ onPress, title, style, textStyle, testID }) => {
  return (
    <Touchable onPress={onPress} testID={testID}>
      <View style={{ ...styles.container, ...style }}>
        <Text style={{ ...styles.text, ...textStyle }} testID="text-output">
          {title}
        </Text>
      </View>
    </Touchable>
  )
}

interface Styles {
  container: ViewStyle
  text: TextStyle
}

const styles = StyleSheet.create<Styles>({
  container: {
    backgroundColor: colors.accentColor,
    paddingHorizontal: 15,
    paddingVertical: 2,
    borderRadius: 4,
  },
  text: {
    color: 'white',
    textTransform: 'uppercase',
    fontFamily: fonts.sansBold,
  },
})

export default Button
