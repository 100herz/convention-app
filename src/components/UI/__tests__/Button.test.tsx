/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react'
import { render } from '@testing-library/react-native'

import Button from '../Button'

const titleMock = 'Button title'

describe('<Button />', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<Button title={titleMock} onPress={() => {}} />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('has no image and renders a View with three texts instead', () => {
    const { getByTestId } = render(<Button title={titleMock} onPress={() => {}} />)
    expect(getByTestId('text-output')).toHaveTextContent(titleMock)
  })
})
