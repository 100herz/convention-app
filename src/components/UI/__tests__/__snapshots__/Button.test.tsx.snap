// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#872232",
        "borderRadius": 4,
        "paddingHorizontal": 15,
        "paddingVertical": 2,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "sans-bold",
          "fontSize": 15,
          "lineHeight": 22.5,
          "textTransform": "uppercase",
        }
      }
      testID="text-output"
    >
      Button title
    </Text>
  </View>
</View>
`;
